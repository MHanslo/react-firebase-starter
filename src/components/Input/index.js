/**
*
* Input Component
*
*/

import React, { PropTypes } from 'react';
import classNames from 'classnames';

function Input(props) {
  const inputClass = classNames('input', {
    'is-danger': props.hasError
  });
  let message = '';
  if (props.hasError) {
    message = (
      <span className="help is-danger">{props.hasError}</span>
    );
  }

  return (
    <div>
      <label className="label">{props.label}</label>
      <div style={{marginBottom: '10px'}}>
        <input
          className={inputClass}
          type={props.type}
          placeholder={props.placeholder}
          value={props.value}
          onChange={props.onChange}
        />
      {message}
      </div>
    </div>
  );
}

Input.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  hasError: PropTypes.string,
  onChange: PropTypes.func
};

export default Input;
