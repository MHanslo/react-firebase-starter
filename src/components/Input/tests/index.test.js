import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../index';

/**
 * Component render
 */
it('should render without errors', () => {
  const component = renderer.create(
    <Input type="text" label="label" placeholder="placeholder" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render an error if hasError is in props', () => {
  const component = renderer.create(
    <Input type="text" hasError="error message" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
