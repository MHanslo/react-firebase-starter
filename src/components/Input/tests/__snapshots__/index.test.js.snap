exports[`test should render an error if hasError is in props 1`] = `
<div>
  <label
    className="label" />
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }>
    <input
      className="input is-danger"
      onChange={undefined}
      placeholder={undefined}
      type="text"
      value={undefined} />
    <span
      className="help is-danger">
      error message
    </span>
  </div>
</div>
`;

exports[`test should render without errors 1`] = `
<div>
  <label
    className="label">
    label
  </label>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }>
    <input
      className="input"
      onChange={undefined}
      placeholder="placeholder"
      type="text"
      value={undefined} />
    
  </div>
</div>
`;
